Client/app/src/main/java/ru/hse/client/SignInActivity.kt

```diff
@@ -0,0 +1,215 @@
+
+
+class SignInActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.sign_in)
+
+        val okHttpClient = OkHttpClient()
+
+        var loginEditText: TextInputEditText = findViewById(R.id.login)
+        var loginLayout: TextInputLayout = findViewById(R.id.login_box)
+        loginEditText.setOnFocusChangeListener{ v, hasFocus ->
+            if (!hasFocus) {
+                hideKeyboard(v)
+            }
+        }
+
+        val textWatcherForLogin: TextWatcher = object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            }
+
+            override fun onTextChanged(
+                charSequence: CharSequence, start: Int, before: Int, count: Int
+            ) {
+                val newText = charSequence.toString()
+                if (!isValidLogin(newText)) {
+                    loginLayout.error = "Unacceptable symbols in login"
+                } else {
+                    // TODO: get login from db and check
+                    loginLayout.boxStrokeColor =
+                        ContextCompat.getColor(this@SignInActivity, R.color.green)
+                    loginLayout.error = null
+                }
+            }
+
+            override fun afterTextChanged(editable: Editable) {
+            }
+        }
+        loginEditText.addTextChangedListener(textWatcherForLogin)
+
+        var passwordEditText: TextInputEditText = findViewById(R.id.password)
+        var passwordLayout: TextInputLayout = findViewById(R.id.password_box)
+        passwordEditText.setOnFocusChangeListener{ v, hasFocus ->
+            if (!hasFocus) {
+                hideKeyboard(v)
+            }
+        }
+
+        val textWatcherForPassword: TextWatcher = object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            }
+
+            override fun onTextChanged(charSequence: CharSequence, start: Int, before: Int, count: Int) {
+                val passwordOriginal = charSequence.toString()
+                if (!isValidPassword(passwordOriginal)) {
+                    passwordLayout.error = "Incorrect password"
+                } else {
+                    passwordLayout.boxStrokeColor = ContextCompat.getColor(this@SignInActivity, R.color.green)
+                    passwordLayout.error = null
+                }
+            }
+
+            override fun afterTextChanged(editable: Editable) {
+            }
+        }
+        passwordEditText.addTextChangedListener(textWatcherForPassword)
+
+        var signInButton: Button = findViewById(R.id.sign_in_button)
+
+        signInButton.setOnClickListener {
+            val login: String = loginEditText.text.toString()
+            val password: String = passwordEditText.text.toString()
+
+            if (!isValidLogin(login)) {
```

Comment:
if this is the main case of using validation functions, then it’s better to name it accordingly: `isNot...`